src/main/kotlin/bitbucket/BitbucketClient.kt

```diff
@@ -26,10 +25,12 @@ class BitbucketClient(
         private val repoSlug: String,
         private val userSlug: String,
         objReader: ObjectReader,
-        private val objWriter: ObjectWriter
+        private val objWriter: ObjectWriter,
+        invalidCredentialsAction: () -> Unit = {}
```

Comment:
The BitbucketClient constructor is getting too big. I think, we could create `responseHandler` in `BitbucketClientFactory`. It wouldn't reduce not so much lines, but we would keep it fit. 

Also it would be great (may be in some other PR) to use `Settings`  from `BitbucketHelperConfigurable` or other data class. It would substitute 4 params in `BitbucketClient` for 1

Reply:
done